tests/backup/backup_longevity_test.go

```diff
@@ -0,0 +1,1567 @@
+		if !ok {
+			chaosLevel = Inst().ChaosLevel
+		}
+		if chaosLevel != 0 {
+			RunningTriggers[triggerType] = triggerInterval[triggerType][chaosLevel]
+		}
+
+	}
+	return nil
+}
+
+// SetTopologyLabelsOnNodes distribute labels on node
+func SetTopologyLabelsOnNodes() ([]map[string]string, error) {
+	// Slice of FA labels
+	topologyLabels := make([]map[string]string, 0)
+	nodeUpTimeout := 5 * time.Minute
+
+	log.Info("Add Topology Labels on node")
+	var secret PureSecret
+	volumeDriverNamespace, err := Inst().V.GetVolumeDriverNamespace()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get volume driver namespace. Error [%v]", err)
+	}
+	pureSecretString, err := Inst().S.GetSecretData(
+		volumeDriverNamespace, PureSecretName, PureSecretDataField,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read pure secret [%s]. Error [%v]",
+			PureSecretName, err)
+	}
+
+	pureSecretJSON := []byte(pureSecretString)
+
+	if err = json.Unmarshal(pureSecretJSON, &secret); err != nil {
+		return nil, fmt.Errorf("Failed to unmarshal pure secret data [%s]. Error:[%v]",
+			pureSecretJSON, err)
+	}
+
+	// Appending the labels to a list
+	for _, fa := range secret.FlashArrays {
+		topologyLabels = append(topologyLabels, fa.Labels)
+	}
+
+	topologyGroups := len(topologyLabels)
+
+	// Adding the labels on node.
+	for nodeIdx, n := range node.GetStorageDriverNodes() {
+		labelIdx := int32(nodeIdx % topologyGroups)
+		for key, value := range topologyLabels[labelIdx] {
+			if err = Inst().S.AddLabelOnNode(n, key, value); err != nil {
+				return nil, fmt.Errorf("Failed to add label key [%s] and value [%s] in node [%s]. Error:[%v]",
+					key, value, n.Name, err)
+			}
+			switch key {
+			case k8s.TopologyZoneK8sNodeLabel:
+				log.InfoD("Setting node: [%s] Topology Zone to: [%s]", n.Name, value)
+				n.TopologyZone = value
+			case k8s.TopologyRegionK8sNodeLabel:
+				log.InfoD("Setting node: [%s] Topology Region to: [%s]", n.Name, value)
+				n.TopologyRegion = value
+			}
+		}
+		// Updating the node with topology info in node registry
+		node.UpdateNode(n)
+	}
+
+	// Bouncing Back the PX pods on all nodes to restart Csi Registrar Container
+	log.Info("Bouncing back the PX pods after setting the Topology Labels on Nodes")
+
+	if err := DeletePXPods(volumeDriverNamespace); err != nil {
+		return nil, fmt.Errorf("Failed to delete PX pods. Error:[%v]", err)
+	}
+
+	// Wait for PX pods to be up
+	log.Info("Waiting for Volume Driver to be up and running")
+	for _, n := range node.GetStorageDriverNodes() {
+		if err := Inst().V.WaitForPxPodsToBeUp(n); err != nil {
+			return nil, fmt.Errorf("PX pod not coming up in a node [%s]. Error:[%v]", n.Name, err)
+		}
+		if err := Inst().V.WaitDriverUpOnNode(n, nodeUpTimeout); err != nil {
+			return nil, fmt.Errorf("Volume Driver not coming up in a node [%s]. Error:[%v]", n.Name, err)
+		}
+	}
+
+	return topologyLabels, nil
+}
+
+func populateIntervals() {
+	triggerInterval = map[string]map[int]time.Duration{}
```

Comment:
Since isTriggerEnabled uses GetWaitTime we need not triggerInterval.

Reply:
please use same functions on longevity test file. else we cant run px and backup events together